docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+
+#### General
+
+The following metrics apply to all (or most) services.
+
+- Service uptime (time since last service boot)
+- Cumulative number API requests succeeded / failed / invalid (2xx vs 5xx vs 4xx)
+- Avg response time (in milliseconds or appropriate unit of measure) on APIs
+- Avg and Max request size
+
+#### Core/Supporting
+
+- Latency (measure of time) an event takes to get through core data
+- Latency (measure of time) a command request takes to get to a device service
+- Indication of health – that events are being processed during a configurable period
+- Number of events in persistence
+- Number of readings in persistence
+- Number of validation failures (validation of device identification)
+- Number of notification transactions
+- Number of notifications handled
+- Number of failed notification transmissions
+- Number of notifications in retry status
+- **CPE** - creation or removal of a new notification subscriber
+- **CPE** - creation or removal of a device profile
+
+#### Application Services
+
+- Processing time for a pipeline; latency (measure of time) an event takes to get through an  application service pipeline
+- DB access times
+- How often are we failing export to be sent to db to be retried at a later time
+- What is the current store and forward queue size
+- How much data (size in KBs or MBs) of packaged sensor data is being sent to an endpoint (or volume)
+- Number of invalid messages that triggered pipeline
+- Number of events processed
+- **CPE** - client endpoint modified
+
+#### Device Services
+
+- **CPE** - device added / provisioned
+- **CPE** - device removed / de-provisioned
+- Number of devices managed by this DS
+- Device Requests (which may be more informative than reading counts and rates)
+
+#### Security
+
+Security metrics may be more difficult to ascertain as they are cross service metrics.  They may have to be dealt with per service.  Also, true threat detection based on metrics may be a feature best provided by 3rd party based on particular threats and security profile needs.
+
+- Number of logins and login failures per service and within a given time
+- Number of secrets accessed per service name
+- Count of any accesses and failures to the data persistence layer
+- Count of service start and restart attempts
+- **CPE** service stop or restart
+- **CPE** registration of new devices and services (allowing for future quarantine mechanism)
+- **CPE** indication of any new uploading of new CA/import to vault
+
+### Design Proposal
+
+- All services will use/integrate go-mod-messaging (if not already integrated).
+- A new metric/CPE model/DTO (Telemetry/Bucket) is required for services to structure the metric or CPE data.  The model/DTO will have similarities to the core event/reading model/DTO.

```

Comment:
Metrics and CPE are timeseries in nature and IMO should stay in timeseries compatible format as it is very likely that this data will be feed into a timeseries DB.